### PR TITLE
introduce MAX_PATH length

### PR DESCRIPTION
**Pull request by dguibert**
It would be sufficient to change the variables, that hold paths and
names to a fixed path length. It has been chosen to set it to 4096
(https://stackoverflow.com/questions/9449241/where-is-path-max-defined-in-linux).

Closes: #9

### CODE REVIEW

File: src/gen_modules_cvmix_idemix.F90 [hunk 1]
```diff
@@ -1,76 +1,76 @@
...
 !
 !===============================================================================
-! module interface to FESOM2.0 for the CVMIX IDEMIX extension for the calculation 
+! module interface to FESOM2.0 for the CVMIX IDEMIX extension for the calculation
 ! of hte internal wave energy and its dissipationof in Turbulent Kinetic Energy
-! vertical mixing scheme --> Its based on the module interface for MPIOM programed 
-! by Nils Brüggeman & Oliver Gutjahr, This module calls subroutines from the 
-! CVMix library 
+! vertical mixing scheme --> Its based on the module interface for MPIOM programed
+! by Nils BrÃ¼ggeman & Oliver Gutjahr, This module calls subroutines from the
```
Comment (koldunovn) [inline src/gen_modules_cvmix_idemix.F90:7]: The Family name is screwed up. I would replace to `Brueggeman`.

File: src/gen_modules_cvmix_idemix.F90 [hunk 2]
```diff
@@ -234,19 +234,19 @@ subroutine init_cvmix_idemix(mesh)
...
             call par_ex(0)
-        end if 
-        
+        end if
+
         !_______________________________________________________________________
-        ! read idemix bottom near tidal forcing from cesm data set --> file 
-        ! from N. Brüggemann interpoalted to regular grid
+        ! read idemix bottom near tidal forcing from cesm data set --> file
+        ! from N. BrÃ¼ggemann interpoalted to regular grid
```
Comment (koldunovn) [inline src/gen_modules_cvmix_idemix.F90:241]: The name.

File: src/gen_modules_cvmix_idemix.F90 [hunk 3]
```diff
@@ -273,54 +273,54 @@ subroutine calc_cvmix_idemix(mesh)
...
         !_______________________________________________________________________
         node_size = myDim_nod2D
         do node = 1,node_size
             nln = nlevels_nod2D(node)-1
-            
+
             !___________________________________________________________________
-            ! calculate for TKE square of Brünt-Väisälä frequency, be aware that
-            ! bvfreq contains already the squared brünt Väisälä frequency ...
+            ! calculate for TKE square of BrÃ¼nt-VÃ¤isÃ¤lÃ¤ frequency, be aware that
```
Comment (koldunovn) [inline src/gen_modules_cvmix_idemix.F90:291]: Name.

File: src/gen_modules_cvmix_idemix.F90 [hunk 4]
```diff
@@ -273,54 +273,54 @@ subroutine calc_cvmix_idemix(mesh)
...
+            ! bvfreq contains already the squared brÃ¼nt VÃ¤isÃ¤lÃ¤ frequency ...
```
Comment (koldunovn) [inline src/gen_modules_cvmix_idemix.F90:292]: Name.

File: src/gen_modules_cvmix_idemix.F90 [hunk 5]
```diff
@@ -343,91 +343,91 @@ subroutine calc_cvmix_idemix(mesh)
...
+            ! |--> FROM NILS: "fac=0.2 ist geschÃ¤tzt. WÃ¼rde ich erstmal so
+            !      probieren. Der kommt aus dem stabilitÃ¤tskriterium fÃ¼r Diffusion
+            !      (Ã¤hnlich berechnet wie das CFL Kriterium nur halt fÃ¼r den
+            !      Diffusions anstatt fÃ¼r den Advektionsterm). Normalerweise
+            !      sollte der Grenzwert aber nicht zu oft auftreten. Ich hatte
+            !      mal damit rum-experimentiert, aber letztendlich war die LÃ¶sung
+            !      das Iterativ zu machen und ggf. idemix_n_hor_iwe_prop_iter zu erhÃ¶hen.
+            !      Du kannst IDEMIX erstmal ohne den Term ausprobieren und sehen,
+            !      ob es lÃ¤uft, dann kannst du den dazuschalten und hoffen, dass
+            !      es nicht explodiert. Eigentlich sollte der Term alles glatter
```
Comment (koldunovn) [inline src/gen_modules_cvmix_idemix.F90:383]: All `umlauts` are destroyed in the paragraph.

File: src/gen_modules_cvmix_idemix.F90 [hunk 6]
```diff
@@ -343,91 +343,91 @@ subroutine calc_cvmix_idemix(mesh)
...
                 do nz=ulevels_nod2D(node)+1,nln
-                    ! inverse volumne 
+                    ! inverse volumne
                     vol_wcelli(nz,node) = 1/(area(nz-1,node)*dz_trr(nz))
-                    
+
                     ! restrict iwe_v0
                     aux = sqrt(cflfac*(area(nz-1,node)/pi*4.0_WP)/(idemix_tau_h*dt/idemix_n_hor_iwe_prop_iter))
-                    !                  `--------+-------------´
+                    !                  `--------+-------------ÃÂ´
```
Comment (koldunovn) [inline src/gen_modules_cvmix_idemix.F90:412]: Extra sighn.

File: src/oce_mesh.F90 [hunk 7]
```diff
@@ -270,13 +270,13 @@ SUBROUTINE read_mesh(mesh)
 
   ! 0 proc reads the data in chunks and distributes it between other procs
   mesh_check=0
-  
-  ! new rotation pole usually (-40.0°,75.0°)
+
+  ! new rotation pole usually (-40.0ÃÂ°,75.0ÃÂ°)
```
Comment (koldunovn) [inline src/oce_mesh.F90:274]: Degrees are destroyed.

File: src/oce_mesh.F90 [hunk 8]
```diff
@@ -295,28 +295,28 @@ SUBROUTINE read_mesh(mesh)
         do n=1, k
            read(fileID,*) ibuff(n,1), rbuff(n,1), rbuff(n,2), ibuff(n,2)
             !___________________________________________________________________
-            ! put here offset parameter so that the checkrotation flag also works 
-            ! for the pi mesh an others there the points around -40 longitude are 
-            ! shifted by 360° --> therefor shift them back with offset parameter
+            ! put here offset parameter so that the checkrotation flag also works
+            ! for the pi mesh an others there the points around -40 longitude are
+            ! shifted by 360ÃÂ° --> therefor shift them back with offset parameter
```
Comment (koldunovn) [inline src/oce_mesh.F90:300]: Again, degrees.

File: src/oce_mesh.F90 [hunk 9]
```diff
@@ -2080,28 +2080,28 @@ SUBROUTINE mesh_auxiliary_arrays(mesh)
...
+!_______________________________________________________________________________
+! calculate gradient/derivative of scalar quantitie via linear shape function
 !                3 (x3,y3)
 !                o
 !               /'\`
 !              /   \
 !             /     \
 !            /       \
-!          '/´        \
+!          '/ÃÂ´        \
```
Comment (koldunovn) [inline src/oce_mesh.F90:2091]: Extra sighn.

File: src/gen_modules_cvmix_idemix.F90 (hunk 1 above)
Comment (dguibert) [reply src/gen_modules_cvmix_idemix.F90:7]: I've restored this changes.

**Comment by dguibert** [comment]
> your editor change a lot of formatting, and mess up some of the letters in the comments (hopefully only there :)). While we should get rid of extra blanks at some point, now is probably not the best time - it makes diffs unreadable, and I expect that it bring more difficulties in merging several upcoming branches (from the point of inspecting them). Would it be possible for you to bring back the formatting and commit only changes related to the path length?

Yes, sorry about that,  I'll push a new commit with only the related changes.

I've commited this on the cluster and there is some UTF-8 issues with `git add -p`
Regarding the whitespaces, I'll try to change my vim config...

BTW for the whitespace I used to have [editorconfig](https://editorconfig.org/) on the repos I work on, this avoid to force anybody to use the same editor but to apply the same configuration via the plugin. Or in some projects, we had [pre-commit](https://pre-commit.com) hooks.

In GitHub "?w=1" could be added for the review: like https://github.com/FESOM/fesom2/pull/65/files?w=1

Those extra blanks could be rid of any time since we could configure git to not bother with space changes during merges (https://stackoverflow.com/a/862920) so I would say the sooner the better ;-)

**Comment by dguibert** [comment]
I've created an `.editorconfig` to keep your formatting style and to only commit the relevant changes, not the formatting ones.
If you are interested I could share it.

**Comment by koldunovn** [comment]
> Yes, sorry about that, I'll push a new commit with only the related changes.
I see you already did it, thanks a lot!

> BTW for the whitespace I used to have [editorconfig](https://editorconfig.org/) on the repos I work on, this avoid to force anybody to use the same editor but to apply the same configuration via the plugin. Or in some projects, we had [pre-commit](https://pre-commit.com) hooks.

That's nice, and I think does not hurt to have it, at least to express some wishes on how the things should be formatted. However I would not put the expectations on many people using it too high, at least initially. The most popular editors that people use for FESOM2 development (AFAIK) currently are `Kate` and `gedit`.
 
> In GitHub "?w=1" could be added for the review: like https://github.com/FESOM/fesom2/pull/65/files?w=1
That's a nice trick, I did not know about it, thanks! I see that there is even a button for this now...

> Those extra blanks could be rid of any time since we could configure git to not bother with space changes during merges (https://stackoverflow.com/a/862920) so I would say the sooner the better ;-)
That should be done on the user side during the merge, right? There is no way to force this behavior in the repository itself?

**Comment by koldunovn** [comment]
> I've created an `.editorconfig` to keep your formatting style and to only commit the relevant changes, not the formatting ones.
> If you are interested I could share it.

Yes, please, that would be nice!

**Comment by koldunovn** [comment]
@dsidoren Dima, can you, please have a look and merge if OK?

**Comment by dguibert** [comment]
> Yes, please, that would be nice!

Done see #66.

**Pull request merged**